dymos/trajectory/trajectory.py

```diff
@@ -417,129 +407,198 @@ def _configure_parameters(self):
+                src_b = options['src_b']
+
+                src_idxs_a = om.slicer[0, ...] if loc_a == 'initial' else om.slicer[-1, ...]
+                src_idxs_b = om.slicer[0, ...] if loc_b == 'initial' else om.slicer[-1, ...]
 
                 if options['connected']:
-                    if var == 'time':
-                        src = '{0}.{1}'.format(phase_name1, source1)
-                        path = 't_initial'
-                        self.connect(src, '{0}.{1}'.format(phase_name2, path))
-                    else:
-                        path = 'initial_states:{0}'.format(var)
-                        self.connect('{0}.{1}'.format(phase_name1, source1),
-                                     '{0}.{1}'.format(phase_name2, path))
-                    print('{3}{0:<{2}s} --> {1:<{2}s}'.format(source1, source2,
-                                                              max_varname_length + 9,
-                                                              indent*2))
+                    if phase_b.classify_var(var_b) == 'time':
+                        self.connect(f'{phase_name_a}.{src_a}',
+                                     f'{phase_name_b}.t_initial',
+                                     src_indices=[-1])
+                        print(f'{indent * 2}{var_a:<{padding}s}  ==  {var_b:<{padding}s}')
+                    elif phase_b.classify_var(var_b) == 'state':
+                        tgt_b = f'initial_states:{var_b}'
+                        self.connect(f'{phase_name_a}.{src_a}',
+                                     f'{phase_name_b}.{tgt_b}',
+                                     src_indices=src_idxs_a)
+                        print(f'{indent * 2}{var_a:<{padding}s}  ->  {var_b:<{padding}s}')
                 else:
+                    linkage_comp.add_linkage_configure(options)
 
-                    self.connect('{0}.{1}'.format(phase_name1, source1),
-                                 'linkages.{0}_{1}:lhs'.format(linkage_name, var))
+                    if f'linkages.{phase_name_a}:{var_a}_{loc_a}' not in connected_linkage_inputs:
+                        self.connect(f'{phase_name_a}.{src_a}',
+                                     f'linkages.{phase_name_a}:{var_a}_{loc_a}',
+                                     src_indices=src_idxs_a)
+                        connected_linkage_inputs.append(f'linkages.{phase_name_a}:{var_a}_{loc_a}')
 
-                    self.connect('{0}.{1}'.format(phase_name2, source2),
-                                 'linkages.{0}_{1}:rhs'.format(linkage_name, var))
+                    if f'linkages.{phase_name_b}:{var_b}_{loc_b}' not in connected_linkage_inputs:
+                        self.connect(f'{phase_name_b}.{src_b}',
+                                     f'linkages.{phase_name_b}:{var_b}_{loc_b}',
+                                     src_indices=src_idxs_b)
+                        connected_linkage_inputs.append(f'linkages.{phase_name_b}:{var_b}_{loc_b}')
```

Comment:
is 'connected_linkage_inputs' actually used for anything once it is assembled?

Reply:
This connect statement is in an if block that checks if the target was already connected.  It's possible to have situations where multiple linkages use the same variables in one or two phases, and this prevents "variable foo is already connected to this input" errors.